File: flux.py
# flux.py

from tokenizer import tokenize
from parser import Parser
from interpreter import Interpreter
import json

with open('index.flux', 'r') as f:
    code = f.read()

tokens = tokenize(code)
parser = Parser(tokens)
ast = parser.parse()

interpreter = Interpreter(ast)
interpreter.run()

print('✅ Execution finished.\n🧠 Environment memory:')
print(json.dumps(interpreter.dump(), indent=2))

File: compiler/flux.js
// flux.js

import { readFileSync } from 'node:fs';
import { tokenize } from './tokenizer.js';
import { Parser } from './parser.js';
import { Interpreter } from './interpreter.js';

const code = readFileSync('index.flux', 'utf-8');

const tokens = tokenize(code);
const parser = new Parser(tokens);
const ast = parser.parse();

const interpreter = new Interpreter(ast);
interpreter.run();

console.log('✅  Execution finished.');
// console.log(JSON.stringify(interpreter.environment, null, 2));
